extension-style/src/test/java/com/mapbox/maps/extension/style/layers/generated/ModelLayerTest.kt

```diff
@@ -0,0 +1,1395 @@
+    layer.modelTranslation(expression)
+    verify { style.setStyleLayerProperty("id", "model-translation", capture(valueSlot)) }
+    assertEquals(valueSlot.captured.toString(), "[+, 2, 3]")
+  }
+
+  @Test
+  fun modelTranslationAsExpressionGet() {
+    val expression = sum {
+      literal(2)
+      literal(3)
+    }
+    every { styleProperty.value } returns TypeUtils.wrapToValue(expression)
+    every { styleProperty.kind } returns StylePropertyValueKind.EXPRESSION
+    val layer = modelLayer("id", "source") { }
+    layer.bindTo(style)
+    assertEquals(expression.toString(), layer.modelTranslationAsExpression?.toString())
+    verify { style.getStyleLayerProperty("id", "model-translation") }
+  }
+
+  @Test
+  fun modelTranslationAsExpressionGetNull() {
+    val layer = modelLayer("id", "source") { }
+    layer.bindTo(style)
+    assertEquals(null, layer.modelTranslationAsExpression)
+    verify { style.getStyleLayerProperty("id", "model-translation") }
+  }
+
+  @Test
+  fun modelTranslationAsExpressionGetFromLiteral() {
+    every { styleProperty.value } returns TypeUtils.wrapToValue(listOf(0.0, 1.0, 2.0))
+    val layer = modelLayer("id", "source") { }
+    layer.bindTo(style)
+    assertEquals("[literal, [0.0, 1.0, 2.0]]", layer.modelTranslationAsExpression.toString())
+    assertEquals(listOf(0.0, 1.0, 2.0), layer.modelTranslation!!)
+    verify { style.getStyleLayerProperty("id", "model-translation") }
+  }
+
+  @Test
+  fun modelTranslationTransitionSet() {
+    val layer = modelLayer("id", "source") {}
+    layer.bindTo(style)
+    layer.modelTranslationTransition(
+      transitionOptions {
+        duration(100)
+        delay(200)
+      }
+    )
+    verify { style.setStyleLayerProperty("id", "model-translation-transition", capture(valueSlot)) }
+    assertEquals(valueSlot.captured.toString(), "{duration=100, delay=200}")
+  }
+
+  @Test
+  fun modelTranslationTransitionGet() {
+    val transition = transitionOptions {
+      duration(100)
+      delay(200)
+    }
+    every { styleProperty.value } returns TypeUtils.wrapToValue(transition)
+    every { styleProperty.kind } returns StylePropertyValueKind.TRANSITION
+    val layer = modelLayer("id", "source") {}
+    layer.bindTo(style)
+    assertEquals(transition.toValue().toString(), layer.modelTranslationTransition?.toValue().toString())
+    verify { style.getStyleLayerProperty("id", "model-translation-transition") }
+  }
+
+  @Test
+  fun modelTranslationTransitionSetDsl() {
+    val layer = modelLayer("id", "source") {}
+    layer.bindTo(style)
+    layer.modelTranslationTransition {
+      duration(100)
+      delay(200)
+    }
+    verify { style.setStyleLayerProperty("id", "model-translation-transition", capture(valueSlot)) }
+    assertEquals(valueSlot.captured.toString(), "{duration=100, delay=200}")
+  }
+
+  @Test
+  fun modelTypeSet() {
+    val layer = modelLayer("id", "source") {}
+    layer.bindTo(style)
+    layer.modelType(ModelType.COMMON_3D)
```

Comment:
maybe good to add `ModelType.LOCATION_INDICATOR` test too.

Reply:
this test is generated, we only generate the first enum at the moment, agree we should probably improve our generator to generate more tests.